src/Stratis.Bitcoin/Configuration/Settings/ConnectionManagerSettings.cs

```diff
@@ -220,4 +238,23 @@ public static void PrintHelp(Network network)
         /// <summary>Filter peers that are within the same IP range to prevent sybil attacks.</summary>
         public bool IpRangeFiltering { get; internal set; }
     }
+
+    public static class ConnectionManagerSettingsExtensions
+    {
+        /// <summary>
+        /// Endpoint A can be mapped to endpoint B.
+        /// </summary>
+        public static bool CanBeMappedTo(this IPEndPoint whiteBindEndpoint, List<IPEndPoint> networkInterfacesTheNodeShouldListenOn, out IPEndPoint localEndpoint)
+        {
+            localEndpoint = networkInterfacesTheNodeShouldListenOn.FirstOrDefault(x => x.Address.IsLocal() && !x.Address.Equals(IPAddress.Any));
```

Comment:
Can we have some comments here explaining how this works please?

Reply:
done